planning/behavior_path_planner/src/utils/avoidance/utils.cpp

```diff
@@ -1137,29 +1137,48 @@ void filterTargetObjects(
 double extendToRoadShoulderDistanceWithPolygon(
   const std::shared_ptr<route_handler::RouteHandler> & rh,
   const lanelet::ConstLineString3d & target_line, const double to_road_shoulder_distance,
-  const geometry_msgs::msg::Point & overhang_pos, const lanelet::BasicPoint3d & overhang_basic_pose)
+  const lanelet::ConstLanelet & overhang_lanelet, const geometry_msgs::msg::Point & overhang_pos,
+  const lanelet::BasicPoint3d & overhang_basic_pose, const bool use_hatched_road_markings,
+  const bool use_intersection_areas)
 {
   // get expandable polygons for avoidance (e.g. hatched road markings)
   std::vector<lanelet::Polygon3d> expandable_polygons;
-  for (const auto & point : target_line) {
-    const auto new_polygon_candidate = utils::getPolygonByPoint(rh, point, "hatched_road_markings");
-    if (!new_polygon_candidate) {
-      continue;
-    }
 
-    bool is_new_polygon{true};
-    for (const auto & polygon : expandable_polygons) {
-      if (polygon.id() == new_polygon_candidate->id()) {
-        is_new_polygon = false;
-        break;
+  if (use_hatched_road_markings) {
+    for (const auto & point : target_line) {
+      const auto new_polygon_candidate =
+        utils::getPolygonByPoint(rh, point, "hatched_road_markings");
+      if (!new_polygon_candidate) {
+        continue;
+      }
+
+      bool is_new_polygon{true};
+      for (const auto & polygon : expandable_polygons) {
+        if (polygon.id() == new_polygon_candidate->id()) {
+          is_new_polygon = false;
+          break;
+        }
+      }
+
+      if (is_new_polygon) {
+        expandable_polygons.push_back(*new_polygon_candidate);
       }
     }
+  }
+
+  if (use_intersection_areas) {
+    const std::string area_id_str = overhang_lanelet.attributeOr("intersection_area", "else");
 
-    if (is_new_polygon) {
-      expandable_polygons.push_back(*new_polygon_candidate);
+    if (area_id_str != "else") {
+      expandable_polygons.push_back(
+        rh->getLaneletMapPtr()->polygonLayer.get(std::atoi(area_id_str.c_str())));
     }
   }
 
+  if (expandable_polygons.empty()) {
+    return to_road_shoulder_distance;
+  }
```

Comment:
```
  auto polygonExistsIn = [](const auto &polygons, const auto &new_polygon) {
    return std::any_of(polygons.begin(), polygons.end(), [&](const auto &polygon){
      return polygon.id() == new_polygon.id();
    });
  };

  if (use_hatched_road_markings) {
    for (const auto & point : target_line) {
      const auto new_polygon_candidate = utils::getPolygonByPoint(rh, point, "hatched_road_markings");

      if (new_polygon_candidate && !polygonExistsIn(expandable_polygons, *new_polygon_candidate)) {
        expandable_polygons.push_back(*new_polygon_candidate);
      }
    }
  }

  if (use_intersection_areas) {
    const std::string area_id_str = overhang_lanelet.attributeOr("intersection_area", "else");

    if (area_id_str != "else") {
      expandable_polygons.push_back(
        rh->getLaneletMapPtr()->polygonLayer.get(std::atoi(area_id_str.c_str())));
    }
  }

  if (expandable_polygons.empty()) {
    return to_road_shoulder_distance;
  }
```

Reply:
I fixed in https://github.com/autowarefoundation/autoware.universe/pull/4206/commits/1e60bcb87a55dab28ac89e8486de728dd2bc6f6c.